src/main/java/com/igot/cb/pores/exceptions/RestExceptionHandling.java

```diff
@@ -0,0 +1,51 @@
+package com.igot.cb.pores.exceptions;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.RestControllerAdvice;
+
+@RestControllerAdvice
+@Slf4j
+public class RestExceptionHandling {
+
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity handleException(Exception ex) {
+        log.debug("RestExceptionHandler::handleException::" + ex);
+        HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
+        ErrorResponse errorResponse = null;
+        if (ex instanceof CustomException) {
+            CustomException CustomException = (CustomException) ex;
+            status = HttpStatus.BAD_REQUEST;
+            // Check if the CustomException provides an HTTP status code
+            if (CustomException != null) {
+                try {
+                    status = CustomException.getHttpStatusCode();
+                } catch (IllegalArgumentException e) {
+                    log.warn("Invalid HTTP status code provided in CustomException: " + CustomException.getHttpStatusCode());
+                }
+            }
```

Comment:
customException is basically ex object -- which is passed in line 15 -- if you are expecting null then it should be checked in line 16

Reply:
This class is not in use so removed